pkg/codegen/go/gen_program_ternaries.go

```diff
@@ -53,8 +55,11 @@ func (ta *tempSpiller) spillExpression(x model.Expression) (model.Expression, hc
 	}, nil
 }
 
-func (g *generator) rewriteTernaries(x model.Expression) (model.Expression, []*ternaryTemp, hcl.Diagnostics) {
-	spiller := &tempSpiller{}
+func (g *generator) rewriteTernaries(
+	x model.Expression,
+	spiller *tempSpiller,
+) (model.Expression, []*ternaryTemp, hcl.Diagnostics) {
+	spiller.temps = nil
```

Comment:
Nice catch!

Reply:
I didn't fix this for spilling in python. Not sure it would cause issues though as reusing variables should be fine in the linear flow of the program.

Reply:
> as reusing variables should be fine in the linear flow of the program.

Yep, that's why we don't do the same thing in Python.